pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -418,7 +418,8 @@ func translateRoute(push *model.PushContext, node *model.Proxy, in *networking.H
 			if action.MaxStreamDuration == nil {
 				action.MaxStreamDuration = &route.RouteAction_MaxStreamDuration{}
 			}
-			action.MaxStreamDuration.MaxStreamDuration = d
+			// Disable MaxStream duration to let the request timeout/gRPC timeout drive the timeout for request.
```

Comment:
MaxStreamDuration is used to reset client connections periodically on running streams https://www.envoyproxy.io/docs/envoy/latest/faq/configuration/timeouts.html?highlight=max_stream_duration#stream-timeouts - Not used for controlling request timeouts. Request timeouts are controlled by request timeout or MaxGrpctimeout (to honour grpc-timeout header capped to this value)

Reply:
I see... we didn't set max stream duration in the past.